src/dataflow/sink/kafka.rs

```diff
@@ -45,69 +46,36 @@ pub fn kafka<G>(
-                            error!(
-                                "error creating topic {} for sink: {}",
-                                connector.topic,
-                                err.to_string()
-                            );
-                            return;
-                        }
-                    };
-                }
-                Err(err) => {
-                    error!(
-                        "error creating new topic {} for sink: {}",
-                        connector.topic,
-                        err.to_string()
-                    );
-                    return;
-                }
-            }
             let producer: FutureProducer = config.create().unwrap();
 
-            stream.sink(Pipeline, &format!("kafka-{}", id), move |input| {
-                input.for_each(|_, rows| {
-                    for (row, _time, diff) in rows.iter() {
-                        let diff_pair = if *diff < 0 {
-                            DiffPair {
-                                before: Some(row),
-                                after: None,
-                            }
-                        } else {
-                            DiffPair {
-                                before: None,
-                                after: Some(row),
+            stream.sink(
+                Exchange::new(move |_| sink_hash),
+                &format!("kafka-{}", id),
+                move |input| {
+                    input.for_each(|_, rows| {
+                        for (row, _time, diff) in rows.iter() {
+                            let diff_pair = if *diff < 0 {
+                                DiffPair {
+                                    before: Some(row),
+                                    after: None,
+                                }
+                            } else {
+                                DiffPair {
+                                    before: None,
+                                    after: Some(row),
+                                }
+                            };
+                            let buf = encoder.encode(schema_id, diff_pair);
+                            for _ in 0..diff.abs() {
+                                producer.send(
+                                    FutureRecord::<&Vec<u8>, _>::to(&connector.topic).payload(&buf),
+                                    1000, /* block_ms */
+                                );
                             }
-                        };
-                        let buf = encoder.encode(schema_id, diff_pair);
-                        for _ in 0..diff.abs() {
-                            producer.send(
-                                FutureRecord::<&Vec<u8>, _>::to(&connector.topic).payload(&buf),
-                                1000, /* block_ms */
-                            );
                         }
-                    }
-                })
-            })
+                    })
+                },
+            )
```

Comment:
It occurs to me that we have the same divergent-dataflow problem with publishing to the schema registry (two lines below). But not your code, so perhaps better to merge this PR as is and fix in a follow up.

Reply:
fixed